File: frontend/bmp-commerce-web/src/types/cliente.ts
export interface Cliente {
  id: string
  nome: string
  cpfCnpj: string | null
  telefone: string | null
  email: string | null
  cidade: string | null
  estado: string | null
  observacoes: string | null
  ativo: boolean
  createdAt: string
  updatedAt: string | null
}

export interface CriarClientePayload {
  nome: string
  cpfCnpj?: string | null
  telefone?: string | null
  email?: string | null
  cidade?: string | null
  estado?: string | null
  observacoes?: string | null
}

export interface AtualizarClientePayload extends CriarClientePayload {
  ativo: boolean
}

export const UFS = [
  'AC', 'AL', 'AP', 'AM', 'BA', 'CE', 'DF', 'ES', 'GO', 'MA', 'MT', 'MS', 'MG',
  'PA', 'PB', 'PR', 'PE', 'PI', 'RJ', 'RN', 'RS', 'RO', 'RR', 'SC', 'SP', 'SE', 'TO',
] as const

File: backend/app/models/cliente.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class ClienteModel(Base):
    __tablename__ = "clientes"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True)
    nome: Mapped[str] = mapped_column(String(200), nullable=False)
    cpf_cnpj: Mapped[str | None] = mapped_column(String(14), nullable=True)
    telefone: Mapped[str | None] = mapped_column(String(30), nullable=True)
    email: Mapped[str | None] = mapped_column(String(256), nullable=True)
    cidade: Mapped[str | None] = mapped_column(String(100), nullable=True)
    estado: Mapped[str | None] = mapped_column(String(2), nullable=True)
    observacoes: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    ativo: Mapped[bool] = mapped_column(Boolean, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
